sp/src/vscript/vscript_squirrel.cpp

```diff
@@ -22,13 +22,15 @@
 #include "sqstdmath.h"
 #include "sqstdstring.h"
 
-// HACK: Include internal parts of squirrel for serialization
```

Comment:
This is still a HACK, in fact it's even a bigger reach into the internals now.

It would be good to look at what can be done to avoid over reaching too far into internals and abstract away what's needed, even if the parts that depend on internals are in their own cpp file.

It's worth also looking to push it up stream to the squirrel lang repo what ever is needed to have proper serialization, this initial version was very hacky and needed to reach into the internals, I did have plans to push more upstream.

Reply:
Implementing complete serialisation natively in Squirrel would be very a large change (and an undertaking), I don't see such feature being accepted any time soon if at all.

Having a substantial serialisation does require accessing the internals anyway. Adding interface to access what is required is also unrealistic since what is required is everything.